Get a new lb pid if existing pid is not alive

jaydoane opened the pull request:
It's possible for the connection process associated with a pid in the
ibrowse_lb ets table to die, yet remain in the table, in which case
subsequent requests to the corresponding `{Host, Port}` will result in an
error like the following:
```
(node1@127.0.0.1)9> ibrowse:send_req("http://localhost:15984", [], get).
                ** exception exit: {noproc,
                       {gen_server,call,
                           [<0.2451.0>,
                            {spawn_connection,
                                {url,"http://localhost:15984","localhost",15984,
                                    undefined,undefined,"/",http,hostname},
                                10,10,
                                {[],false},
                                []}]}}
     in function  gen_server:call/2 (gen_server.erl, line 215)
     in call from ibrowse:try_routing_request/14 (src/ibrowse.erl, line 377)
```
This checks whether the pid about to be returned from the table is alive,
and if not, the entry is deleted, and a new pid is obtained.

This is a backport of https://github.com/cmullaparthi/ibrowse/pull/166

nickva commented on src/ibrowse.erl:
```diff
@@ -335,6 +330,20 @@ send_req(Url, Headers, Method, Body, Options, Timeout) ->
+    case ets:lookup(ibrowse_lb, {Host, Port}) of
+        [] ->
+            get_lb_pid(Url);
+        [#lb_pid{pid = Lb_pid_1}] ->
```
Minor nit: don't need the Lb_pid_1 as there is only one Pid in this function

jaydoane replied:
Ah, good point! I just copied that code block without thinking about simplification. I have updated the PR renaming `Lb_pid_1` to simply `Pid`. Thanks for the eagle eye. I'll update the upstream PR as well.

nickva reviewed (approved):
Nice fix, Jay!

jaydoane committed:
Get a new lb pid if existing pid is not alive

It's possible for the connection process associated with a pid in the
ibrowse_lb ets table to die, yet remain in the table, in which case
subsequent requests to the corresponding {Host, Port} will result in an
error like the following:

(node1@127.0.0.1)9> ibrowse:send_req("http://localhost:15984", [], get).
                ** exception exit: {noproc,
                       {gen_server,call,
                           [<0.2451.0>,
                            {spawn_connection,
                                {url,"http://localhost:15984","localhost",15984,
                                    undefined,undefined,"/",http,hostname},
                                10,10,
                                {[],false},
                                []}]}}
     in function  gen_server:call/2 (gen_server.erl, line 215)
     in call from ibrowse:try_routing_request/14 (src/ibrowse.erl, line 377)

This checks whether the pid about to be returned from the table is alive,
and if not, the entry is deleted, and a new pid is obtained.

wohali commented:
@jaydoane don't forget to tag here, then bump `rebar.config.script` in the main repo to pull this in!

jaydoane commented:
@wohali thanks for the reminder! Do you suggest using `CouchDB-4.0.2` for the tag?

wohali commented:
@jaydoane No, because it doesn't track the upstream 4.0.2 version, and it doesn't include the 4.3 version (see https://github.com/apache/couchdb/pull/437 for why that hasn't happened yet).

What you want is something like `CouchDB-4.0.1-1`.

jaydoane commented:
@wohali got it, thanks. I've pushed the new tag, and here's 
a PR to bump ibrowse in rebar.config.script: https://github.com/apache/couchdb/pull/2001
